scalalib/src/mill/scalalib/bsp/MillBuildModule.scala

```diff
@@ -1,131 +0,0 @@
-    BuildInfo.millBinPlatform
-  }
-
-  override def compileIvyDeps: T[Agg[Dep]] = T {
-    val deps = Agg.from(BuildInfo.millEmbeddedDeps.split(",").map(d => ivy"${d}".forceVersion()))
-    T.log.errorStream.println(s"compileIvyDeps: ${deps}")
-    deps
-  }
-
-  override def ivyDeps: T[Agg[Dep]] = T {
-    val deps = buildScImports().collect {
-      case d: IncludedDep =>
-//        val Seq(dep) = MillIvy.processMillIvyDepSignature(Seq(d.dep))
-        val Seq(dep) = Seq(d.dep)
-        ivy"${dep}"
-    }
-    T.log.errorStream.println(s"ivyDeps: ${deps}")
-    Agg.from(deps)
-  }
-
-  override def scalacPluginIvyDeps: Target[Loose.Agg[Dep]] = T {
-    val deps = BuildInfo.millScalacPluginDeps.map(d => ivy"${d}")
-    T.log.errorStream.println(s"scalacPluginIvyDeps: ${deps}")
-    deps
-  }
-
-  /**
-   * We need to add all resources from Ammonites cache,
-   * which typically also include resolved `ivy`-imports and compiled `$file`-imports.
-   */
-//  override def unmanagedClasspath: T[Agg[PathRef]] = T {
-//    super.unmanagedClasspath() ++ (
-//      rootModule.getClass.getClassLoader match {
-//        case cl: SpecialClassLoader =>
-//          cl.allJars.map(url => PathRef(os.Path(java.nio.file.Paths.get(url.toURI))))
-//        case _ => Seq()
-//      }
-//    )
-//  }
-
-  // We need to be careful here to not include the out/ directory
-  override def sources: Sources = T.sources {
-    val sources = allSourceFiles()
-    T.log.errorStream.println(s"sources: ${sources}")
-    sources
-  }
-
-  override def allSourceFiles: T[Seq[PathRef]] = T {
-    val buildSc = buildScFile()
-    val includedFiles = buildScImports().collect {
-      case i: IncludedFile => i.file
-    }.map(p => PathRef(p))
-    val all = buildSc ++ includedFiles
-    T.log.errorStream.println(s"allSourceFiles: ${all}")
-    all
-  }
-
-  override def compileResources: Sources = T.sources()
-  override def resources: Sources = T.sources()
-
-  override def bspBuildTarget: BspBuildTarget = super.bspBuildTarget.copy(
-    displayName = Some("mill-build"),
-    baseDirectory = Some(projectPath),
-    languageIds = Seq(BspModule.LanguageId.Scala),
-    canRun = false,
-    canCompile = false,
-    canTest = false,
-    canDebug = false,
-    tags = Seq(BspModule.Tag.Library, BspModule.Tag.Application)
-  )
-  override def compile: T[CompilationResult] = T {
-    T.log.errorStream.println(s"compile: ${T.dest}")
-    os.write(T.dest / "dummy", "")
-    os.makeDir(T.dest / "classes")
-    CompilationResult(T.dest / "dummy", PathRef(T.dest / "classes"))
-  }
-
-  override def semanticDbData: T[PathRef] = T {
-    T.log.errorStream.println(s"semanticDbData: ${T.dest}")
-    PathRef(T.dest)
-  }
-
-  /** Used in BSP IntelliJ, which can only work with directories */
-  def dummySources: Sources = T.sources(T.dest)
-}
-
-/**
- * Just for testing purposes.
- */
-@internal
-object MillBuildModule extends ExternalModule with MillBuildModule {
```

Comment:
@lihaoyi Is there any way after this change, to inspect a root module, e.g. to show it's sources etc.?

Reply:
For `RootModule`, you can just `mill show sources`. For `MillBuildRootModule`, there wasn't a way before this PR and this PR doesn't change things AFAIK. We'd need to add it in another PR.

Reply:
Ah thanks. I thought, I tested it, but probably with the wrong mill dev version. 🤦‍♂️

I see `src` included, but not the `build.sc`. This is probably the reason, you don't get language-specific edit features for `build.sc`.